### PR TITLE
MM-T254 Rehydrate mention badge after post is marked as Unread

### PR DESCRIPTION
**Pull request by openmohan**
#### Summary

MM-T254 Rehydrate mention badge after post is marked as Unread

#### Ticket Link

https://automation-test-cases.vercel.app/test/MM-T254

### CODE REVIEW

File: e2e/cypress/integration/mark_as_unread/unread_toast_count_spec.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        // # Mark the first post as unread
+        cy.getNthPostId(1).then((postId) => {
+            markAsUnreadByPostIdFromMenu(postId);
+        });
+
+        // * Check that the toast is visible
+        cy.get('div.toast').should('be.visible').then(() => {
+            // * Check that the message is correct and all the mentions and replies are counted in toast
+            cy.get('div.toast__message>span').should('be.visible').first().contains('32 new messages');
+        });
```
Comment (srkgupta) [inline e2e/cypress/integration/mark_as_unread/unread_toast_count_spec.js:66]: The test is missing the following assertion:
`In particular, at-here should mention the user even if they were offline when the post was originally made.`

Suggested changes:
```
        cy.get(`#sidebarItem_${currentChannel.name}`).
            scrollIntoView().
            find('#unreadMentions').
            should('be.visible').
            and('have.text', '1');
```

Note: You need to declare and assign the variable currentChannel before.

**Review by srkgupta — CHANGES_REQUESTED** [review]
Hi @openmohan 
Thanks. The test looks great. Just one suggestion observed.

**Review by srkgupta — APPROVED** [review]
Thanks for the changes @openmohan. The changes looks good. Approving the PR.

File: e2e/cypress/integration/mark_as_unread/unread_toast_count_spec.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                        });
+                    });
+
+                    // # Make a mentioned message
+                    cy.postMessageAs({
+                        sender: user2,
+                        message: `hi @${user.username}`,
+                        channelId: channel.id,
+
+                    });
```
Comment (saturninoabril) [inline e2e/cypress/integration/mark_as_unread/unread_toast_count_spec.js:40]:
```suggestion
                        channelId: channel.id,
                    });
```

File: e2e/cypress/integration/mark_as_unread/unread_toast_count_spec.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                        cy.clickPostCommentIcon(postId);
+                        cy.get('#rhsContainer').should('be.visible');
+                        const replyMessage = 'A reply to an older post';
+                        cy.postMessageReplyInRHS(replyMessage);
+                    });
+                });
+            });
+        });
+    });
+    it('MM-T254 Rehydrate mention badge after post is marked as Unread', () => {
```
Comment (saturninoabril) [inline e2e/cypress/integration/mark_as_unread/unread_toast_count_spec.js:59]: nit,
```suggestion
    });

    it('MM-T254 Rehydrate mention badge after post is marked as Unread', () => {
```

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @openmohan! Tested and passed. LGTM, except for minor suggestions.

**Comment by openmohan** [comment]
Thanks, @saturninoabril for the review and suggestions.

**Review by saturninoabril — APPROVED** [review]
Thanks @openmohan! LGTM

**Pull request merged**